cluster_management/src/main/java/com/pinterest/rocksplicator/Utils.java

```diff
@@ -60,6 +63,42 @@
   private static final String RESOURCE_INFO_META_SUFFIX = "resource_meta";
   private static final String RESOURCE_INFO_CONFIGS_SUFFIX = "resource_configs";
 
+  /**
+   * Obtain the Helix External View of a resource in a cluster.
+   * It performs a fixed number of retries to mitigate potential transient failure.
+   * @param admin Helix Admin client
+   * @param clusterName
+   * @param resourceName
+   * @return ExternalView, which will be null if failed
+   */
+  public static ExternalView getHelixExternalViewWithFixedRetry(HelixAdmin admin, String clusterName, String resourceName) {
+    ExternalView view = admin.getResourceExternalView(clusterName, resourceName);
+    if (view != null) {
+      return view;
+    }
+
+    LOG.error("Failed to get external view for " + resourceName + ", retry ...");
+
+    // retry for maxRetries times with sleep in between
+    final int maxRetries = 3;
+    final int retryWaitSec = 1;
+    try {
+      for (int retryCount = 1; retryCount <= maxRetries; retryCount++) {
+        TimeUnit.SECONDS.sleep(retryWaitSec * retryCount);
```

Comment:
Should we do exponential backoff retry?

Reply:
Updated, though I think it doesn't matter in this case, exponentially backoff is typically only used for retrying for a long period (to avoid retrying too frequently and causing issues)